client/db.go

```diff
@@ -419,6 +419,19 @@ func (db *DB) Txn(retryable func(txn *Txn) error) error {
 	return newTxn(*db, 1 /* depth */).exec(retryable)
 }
 
+// BeginTxn starts a new transaction.
+func (db *DB) BeginTxn() *Txn {
+	return newTxn(*db, 1 /* depth */)
+}
+
+// ContinueTxn returns a transaction created from the preserved
+// state.
+func (db *DB) ContinueTxn(state proto.Transaction) *Txn {
```

Comment:
same as above, client/db shouldn't know about this; `NewTxn()` should give you everything you need